src/productcatalogservice/README.md

```diff
@@ -3,3 +3,23 @@
 Run the following command to restore dependencies to `vendor/` directory:
 
     dep ensure --vendor-only
+
+## Triggering a bug in ProductCatalogService
+
+This service has an embedded bug in the `parseCatalog` function. It slows down
+the whole application and consumes some CPU cycles. It will show up in profiling
+tools.
+
+You can trigger this bug by sending a `USR1` signal and remove it (if needed) by
+sending a `USR2` signal:
+
+```
+# Trigger bug
+kubectl exec \
+    $(kubectl get pods -l app=productcatalogservice -o jsonpath='{.items[0].metadata.name}') \
+    -c server -- kill -USR1 1
```

Comment:
We're about to add a entrypoint wrapper for Stackdriver Debugger to all these processes in #87 . I am not sure the debugger process is able to forward signals to its child process adequately. Do you mind testing?

Reply:
Will do and report back.

Reply:
Thanks, that'll be great. I am sorry this is very disrupting.

Reply:
So, I'm having a hard time figuring out what's happening here:
- The debugger seems to slow down significatively productcatalogservice in any case.
- It also seems to break the logging: I'm getting the first few lines of logs of productcatalogservice, but then nothing, even if I send directly a USR1 or USR2 to the productcatalogservice PID.
- I looked a little bit at the [code of the debugger](https://github.com/GoogleCloudPlatform/google-cloud-go/tree/master/cmd/go-cloud-debug-agent), and it doesn't look like it's forwarding USR1 and USR2. 

We can work around the fact that the debugger doesn't forward the signals, but the slowness introduced and the fact that we don't have logs anymore is a problem.

I'll open a couple of issues in [google-cloud-go](https://github.com/GoogleCloudPlatform/google-cloud-go).